test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -120,12 +133,57 @@ UCS_TEST_P(test_uct_sockaddr, connect_client_to_server)
      * test ends and the client's ep was destroyed */
 }
 
+UCS_TEST_P(test_uct_sockaddr, connect_client_to_server_with_delay)
+{
+    UCS_TEST_MESSAGE << "Testing " << ucs::sockaddr_to_str(listen_sock_addr.addr)
+                     << " Interface: " << GetParam()->dev_name.c_str();
+    delay_conn_reply = true;
+    client->connect(0, *server, 0, &connect_sock_addr);
+
+    /* wait for the server to connect */
+    while (server_recv_req == 0) {
+        progress();
+    }
+    ASSERT_EQ(1, server_recv_req);
+    ASSERT_EQ(1ul, delayed_conn_reqs.size());
+    EXPECT_EQ(0, err_count);
+    while (!delayed_conn_reqs.empty()) {
+        uct_iface_accept(server->iface(), delayed_conn_reqs.front());
+        delayed_conn_reqs.pop();
+    }
+    sleep(3);
```

Comment:
why sleep needed? can we have a condition check instead?

Reply:
Copy/pasted from https://github.com/evgeny-leksikov/ucx/pull/3/files#diff-93fd01f353a7644c48f98fd17ae69187R277
I guess condition check requires transport level hooks